tf/getter.go

```diff
@@ -135,18 +152,30 @@ func (tfrGetter *RegistryGetter) Get(dstPath string, srcURL *url.URL) error {
 		return errors.New(MalformedRegistryURLErr{reason: "more than one version query"})
 	}
 
-	version := versionList[0]
+	versionQuery := versionList[0]
 
 	moduleRegistryBasePath, err := GetModuleRegistryURLBasePath(ctx, tfrGetter.TerragruntOptions.Logger, registryDomain)
 	if err != nil {
 		return err
 	}
 
-	moduleURL, err := BuildRequestURL(registryDomain, moduleRegistryBasePath, modulePath, version)
+	moduleVersionsURL, err := BuildModuleVersionsURL(registryDomain, moduleRegistryBasePath, modulePath)
+	if err != nil {
+		return err
+	}
+
+	targetVersion, err := GetTargetVersion(ctx, tfrGetter.TerragruntOptions.Logger, *moduleVersionsURL, versionQuery)
+	if err != nil {
+		return err
+	}
+
+	moduleURL, err := BuildRequestURL(registryDomain, moduleRegistryBasePath, modulePath, targetVersion)
 	if err != nil {
 		return err
 	}
 
+	tfrGetter.TerragruntOptions.Logger.Infof("Downloading module from %s", moduleURL.String())
```

Comment:
I'm not entirely sure we need this log message at the Info level. It may produce multiple unused lines, so would it make more sense to set it to Debug instead?

Reply:
I have doubts also. However, since there's another Info log related to this and it includes the query string, I considered useful to have now this second log line clarifying which version is finally being downloaded without having to go some levels up (way far from getter.go) to change that behaviour.

As an example:
![image](https://github.com/user-attachments/assets/34a0397e-4ae1-4855-bedb-8f3d60bde078)

That said, I believe this is an intermediate step. The next one should be to implement the version attribute and properly handle the version constraints at an upper level (adding the ability to invalidate tg cache when there's a new version even if the code hasn't changed, among other changes). Then, both log lines should be addressed and merged into an more accurate one. WDYT?

Reply:
FYI: Since the scope of the PR has been changed to also support a new version attribute, and this will imply modifying way more than the tf getter, I believe this logs issue could be addressed also at the root, so this log line in getter won't make sense anymore